test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetIntegrationTestFixture.cs

```diff
@@ -0,0 +1,506 @@
+            => RunDotnet(workingDirectory, args, expectSuccess: false, environmentVariables);
+
+        private CommandRunnerResult RunDotnet(string workingDirectory, string args = "", bool expectSuccess = true, IReadOnlyDictionary<string, string> environmentVariables = null)
+        {
+            bool enableDiagnostics = CIDebug && !string.IsNullOrWhiteSpace(BinLogDirectory);
+
+            FileInfo coreHostLogFileInfo = null;
+
+            Dictionary<string, string> finalEnvironmentVariables = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                ["MSBuildSDKsPath"] = MsBuildSdksPath,
+                ["UseSharedCompilation"] = bool.FalseString,
+                ["DOTNET_MULTILEVEL_LOOKUP"] = "0",
+                ["DOTNET_ROOT"] = _cliDirectory,
+                ["MSBUILDDISABLENODEREUSE"] = bool.TrueString
+            };
+
+            if (enableDiagnostics)
+            {
+                coreHostLogFileInfo = new FileInfo(Path.GetTempFileName());
+                coreHostLogFileInfo.Delete();
+
+                finalEnvironmentVariables["COREHOST_TRACE"] = "1";
+                finalEnvironmentVariables["COREHOST_TRACEFILE"] = coreHostLogFileInfo.FullName;
+                finalEnvironmentVariables["COMPlus_DbgEnableElfDumpOnMacOS"] = "1";
+                finalEnvironmentVariables["COMPlus_DbgEnableMiniDump"] = "1";
+                finalEnvironmentVariables["COMPLUS_DBGENABLEMINIDUMP"] = "1";
+                finalEnvironmentVariables["COMPlus_DbgMiniDumpName"] = Path.Combine(BinLogDirectory, $"minidump-%p-%t.dmp");
+                finalEnvironmentVariables["COMPLUS_DBGMINIDUMPNAME"] = Path.Combine(BinLogDirectory, $"minidump-%p-%t.dmp");
+                finalEnvironmentVariables["COMPlus_DbgMiniDumpType"] = "4";
+                finalEnvironmentVariables["COMPLUS_DBGMINIDUMPTYPE"] = "4";
+                finalEnvironmentVariables["DOTNET_DbgEnableMiniDump"] = "1";
+                finalEnvironmentVariables["DOTNET_DbgMiniDumpName"] = Path.Combine(BinLogDirectory, $"minidump-%p-%t.dmp");
+            }
+
+            if (environmentVariables != null)
+            {
+                foreach (var item in environmentVariables)
+                {
+                    finalEnvironmentVariables[item.Key] = item.Value;
+                }
+            }
+
+            CommandRunnerResult result = null;
+
+            try
+            {
+                Stopwatch stopwatch = Stopwatch.StartNew();
+
+                result = CommandRunner.Run(TestDotnetCli, workingDirectory, args, environmentVariables: finalEnvironmentVariables);
+
+                stopwatch.Stop();
+
+                if (expectSuccess)
+                {
+                    Assert.True(result.ExitCode == 0, $"{TestDotnetCli} {args} should have succeeded but returned exit code {result.ExitCode} after {stopwatch.Elapsed.TotalSeconds:N1}s with the following output:{Environment.NewLine}{result.AllOutput}");
```

Comment:
Use FluentAssertions to get the full error message.

The Xunit asserts truncate.

Reply:
Done!